File: problems/2375_construct_smallest_number_from_di_string/Solution.java
import java.util.*;

class Solution {
    public String smallestNumber(String pattern) {
        StringBuilder result = new StringBuilder();
        Deque<Character> stack = new ArrayDeque<>();
        for (int i = 0; i <= pattern.length(); i++) {
            stack.push((char)('1' + i));
            if (i == pattern.length() || pattern.charAt(i) == 'I') {
                while (!stack.isEmpty()) result.append(stack.pop());
            }
        }
        return result.toString();
    }
}

File: problems/2375_construct_smallest_number_from_di_string/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    string smallestNumber(string pattern) {
        string result;
        stack<char> st;
        for (int i = 0; i <= pattern.size(); i++) {
            st.push('1' + i);
            if (i == pattern.size() || pattern[i] == 'I') {
                while (!st.empty()) { result += st.top(); st.pop(); }
            }
        }
        return result;
    }
};
